sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ColumnarBatch.java

```diff
@@ -307,64 +293,70 @@ public void update(int ordinal, Object value) {
 
     @Override
     public void setNullAt(int ordinal) {
```

Comment:
It seems like the rows returned by `ColumnarBatch.rowIterator` doesn't need to be mutable with our current tests, but `ColumnarBatch.Row` still needs to be mutable, the write apis of which are used in `HashAggregateExec`.